services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouteController.java

```diff
@@ -45,40 +49,69 @@
     @Autowired
     private RouteDatabaseService routeDatabaseService;
 
+    @Autowired
+    private RouteWithVpcMapperService routeWithVpcMapperService;
+
+    @Autowired
+    private RouteWithSubnetMapperService routeWithSubnetMapperService;
+
     @RequestMapping(
             method = GET,
-            value = {"/vpcs/{vpcId}/routes/{routeId}"})
-    public RouteWebJson getRuleByVpcId(@PathVariable String vpcId, @PathVariable String routeId) throws Exception {
+            value = {"routes/vpcs/{vpcId}/get"})
+    public RoutesWebJson getRulesByVpcId(@PathVariable String vpcId) throws Exception {
 
-        RouteEntity routeEntity = null;
+        List<RouteEntity> routes = null;
 
         try {
             RestPreconditionsUtil.verifyParameterNotNullorEmpty(vpcId);
-            RestPreconditionsUtil.verifyParameterNotNullorEmpty(routeId);
 
-            routeEntity = this.routeDatabaseService.getByRouteId(routeId);
+            routes = this.routeWithVpcMapperService.getRuleByVpcId(vpcId);
         } catch (ParameterNullOrEmptyException e) {
             //TODO: REST error code
             throw new Exception(e);
         }
 
-        if (routeEntity == null) {
+        if (routes == null) {
             //TODO: REST error code
-            return new RouteWebJson();
+            return new RoutesWebJson();
         }
 
-        return new RouteWebJson(routeEntity);
+        return new RoutesWebJson(routes);
     }
 
     @RequestMapping(
             method = GET,
-            value = {"/subnets/{subnetId}/routes/{routeId}"})
-    public RouteWebJson getRuleBySubnetId(@PathVariable String subnetId, @PathVariable String routeId) throws Exception {
+            value = {"routes/subnets/{subnetId}/get"})
+    public RoutesWebJson getRulesBySubnetId(@PathVariable String subnetId) throws Exception {
 
-        RouteEntity routeEntity = null;
+        List<RouteEntity> routes = null;
 
         try {
             RestPreconditionsUtil.verifyParameterNotNullorEmpty(subnetId);
+
+            routes = this.routeWithSubnetMapperService.getRuleBySubnetId(subnetId);
+        } catch (ParameterNullOrEmptyException e) {
```

Comment:
Let us change the route manager exception handling in this PR. 

One idea is to define different exception (each with specific error codes and message), and we could throw a specific exception instead of a general exception.

An example is Port Manager which defines 20+ different exceptions for various corner cases.
https://github.com/futurewei-cloud/alcor/tree/master/services/port_manager/src/main/java/com/futurewei/alcor/portmanager/exception

Reply:
Got it